packages/gatsby/cache-dir/dev-loader.js

```diff
@@ -75,6 +77,9 @@ class DevLoader extends BaseLoader {
   }
 
   doPrefetch(pagePath) {
+    if (process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND) {
+      return Promise.resolve()
+    }
```

Comment:
Will this also affect hover over a link?

Reply:
Heh, I actually inline commented on this in tracking branch - 

> We call this whenever link is in viewport - this makes dev server / runtime feel unresponsive when there is a lot of heavy pages linked.
>
> We will continue to preload resources on hovering for now - it does have similar potential to cause unresponsiveness if someone flicks their mouse through list of links (i.e. horizontal menu bars or vertical sidebar with bunch of links), but overall hover behaviour seems safer to keep because we can start running queries just a bit sooner than if we would wait for a click making delay in loading page feel a bit faster (perceived perf is magic btw :) )

But to add to it - `DevLoader` is subclass of `BaseLoader` which implements `hovering` method - https://github.com/gatsbyjs/gatsby/blob/23b413741e0ea4079752a21f0c390ae590d5bc26/packages/gatsby/cache-dir/loader.js#L383-L385

And because it's not overwritten - both `DevLoader` and `ProdLoader` have the same behaviour of fetching `page-data` files

Reply:
I think this is subject to changes depending on feedback we get ... and possibly we can make this configurable - like `process.env.GATSBY_EXPERIMENTAL_QUERY_ON_DEMAND=only-navigation` (etc)

The main problem to me is image processing so once we can ensure we have lazy image processing we could be more confident in keeping the behaviour of hovers preloading the resources